runtime/src/rpc.rs

```diff
@@ -209,44 +206,77 @@ impl InterBtcParachain {
         // For getting the nonce, use latest, possibly non-finalized block.
         // TODO: we might want to wait until the latest block is actually finalized
         // query account info in order to get the nonce value used for communication
+        let storage_key = metadata::storage().system().account(&self.account_id);
         let account_info = self
             .api
             .storage()
-            .system()
-            .account(&self.account_id, None)
+            .fetch(&storage_key, None)
             .await
+            .transpose()
+            .map(|x| x.ok())
+            .flatten()
             .map(|x| x.nonce)
-            .unwrap_or(0);
+            .unwrap_or_default();
 
         log::info!("Refreshing nonce: {}", account_info);
         signer.set_nonce(account_info);
     }
 
-    /// Gets a copy of the signer with a unique nonce
-    async fn with_unique_signer<'client, F, R>(
+    async fn query_latest<Address>(
         &self,
-        call: F,
-    ) -> Result<TransactionEvents<'client, InterBtcRuntime, InterBtcEvent>, Error>
+        address: Address,
+    ) -> Result<Option<<Address::Target as DecodeWithMetadata>::Target>, Error>
     where
-        F: Fn(InterBtcSigner) -> R,
-        R: Future<
-            Output = Result<TransactionProgress<'client, InterBtcRuntime, DispatchError, InterBtcEvent>, BasicError>,
-        >,
+        Address: StorageAddress<IsFetchable = Yes>,
+    {
+        let hash = self.get_latest_block_hash().await?;
```

Comment:
It seems to me that you're now always fetching the latest block, rather than the latest _finalized_ block. Or did something change under the hood of subxt?

Reply:
This is just poor naming, let me rename the function signatures.